apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -50,7 +50,8 @@ type APIGroup struct {
 	*common.PasswordChanger
 	*common.LifeGetter
 	*common.AgentEntityWatcher
-	*charmscommon.CharmsAPI
+	*charmscommon.CharmInfoAPI
+	*charmscommon.ApplicationCharmInfoAPI
```

Comment:
We don't want to embed common API facades any more. It makes for a versioning nightmare. The preferred technique is to set it as a member and wrap the calls.

Reply:
Makes sense. That's part of why I renamed `CharmsAPI` to `CharmInfoAPI` to clearly show which API endpoint you're embedding (and only one). But still, it's more explicit with regular fields -- I'll update to that approach in a bit.